src/pages/workspace/NewWorkspacePage.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import {View, Text} from 'react-native';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import Navigation from '../../libs/Navigation/Navigation';
+import styles from '../../styles/styles';
+import WorkspaceDefaultAvatar from '../../../assets/images/workspace-default-avatar.svg';
+import TextInputWithLabel from '../../components/TextInputWithLabel';
+import Button from '../../components/Button';
+import AttachmentPicker from '../../components/AttachmentPicker';
+import Icon from '../../components/Icon';
+import {DownArrow, Upload} from '../../components/Icon/Expensicons';
+import CreateMenu from '../../components/CreateMenu';
+import Switch from '../../components/Switch';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+};
+
+class NewWorkspacePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isEditPhotoMenuVisible: false,
+            name: '',
+            requestCall: false,
+        };
+
+        this.createMenuItems = this.createMenuItems.bind(this);
+    }
+
+    /**
+     * Create menu items list for avatar menu
+     *
+     * @param {Function} openPicker
+     * @returns {Array}
+     */
+    createMenuItems(openPicker) {
+        const menuItems = [
+            {
+                icon: Upload,
+                text: this.props.translate('workspace.new.uploadPhoto'),
+                onSelected: () => {
+                    openPicker({
+                        onPicked: () => {
+                            // TODO: connect with setWorkspaceAvatar function
+                        },
+                    });
+                },
+            },
+        ];
+
+        // TODO: Add option to remove avatar if the user doesn't like the one they chose.
+
+        return menuItems;
+    }
+
+    render() {
+        const {translate} = this.props;
```

Comment:
NAB, IIRC we have some kind of specific advice on when to destructure something in the style guide. but generally prefer explicit use of `this.props` when there is a mix of props and state to avoid confusion about which things come from where. That said, t's quite easy to see what's happening here so I won't block.

Reply:
Done!